pageserver/src/tenant/vectored_blob_io.rs

```diff
@@ -509,8 +529,11 @@ mod tests {
         planner.handle_range_end(652 * 1024);
 
         let reads = planner.finish();
+
         assert_eq!(reads.len(), 6);
 
+        // TODO: should we produce 5 reads instead? Currently, the last read has the same start=end=652*1024, a zero-sized read.
```

Comment:
I think this is fine. It's an artefact of the last blob starting and ending at offset 652 * 1024 (see `handle_range_end` above). This is not a valid input, but we could extend `VectoredRead::finish` to remove any zero sized reads.

Reply:
updated the comments